oak_utils/src/lib.rs

```diff
@@ -204,3 +204,42 @@ where
         .compile_protos(inputs, includes)
         .expect("could not run prost-build");
 }
+
+/// Options for building gRPC code.
+pub struct CodegenOptions {
+    /// Specify whether to build client related code.
+    pub build_client: bool,
+    /// Specify whether to build server related code.
+    pub build_server: bool,
+}
+
+/// Generate gRPC code from Protobuf using `tonic` library.
+pub fn generate_grpc_code(
+    proto_path: &str,
+    file_paths: &[&str],
+    options: CodegenOptions,
+) -> std::io::Result<()> {
+    // TODO(#1093): Move all proto generation to a common crate.
+    let proto_path = std::path::Path::new(proto_path);
+    let file_paths: Vec<std::path::PathBuf> = file_paths
+        .iter()
+        .map(|file_path| proto_path.join(file_path))
+        .collect();
+    // Tonic compile requires a vector of `Path` references.
+    let file_path_refs: Vec<&std::path::Path> = file_paths
+        .iter()
+        .map(|file_path| {
+            // Tell cargo to rerun this build script if the proto file has changed.
+            // https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath
+            println!("cargo:rerun-if-changed={}", file_path.display());
```

Comment:
`map` is lazily evaluated, my impression is that side effects inside `map` are discouraged. See for example https://internals.rust-lang.org/t/add-foreach-method-to-iterators-for-side-effects/1312.

Maybe you can just put the `println` in a separate for loop?

Reply:
Done